rotkehlchen/tests/unit/test_tasks_manager.py

```diff
@@ -221,21 +222,37 @@ def test_check_premium_status(rotkehlchen_api_server):
     premium = Premium(premium_credentials)
     premium.status = SubscriptionStatus.ACTIVE
 
+    def mock_check_premium_status():
+        task_manager.last_premium_status_check = ts_now() - PREMIUM_STATUS_CHECK
+        task_manager._maybe_check_premium_status()
+
     with patch(
         'rotkehlchen.db.dbhandler.DBHandler.get_rotkehlchen_premium',
         MagicMock(return_value=premium_credentials),
     ):
         assert premium.is_active() is True
         assert rotki.premium is not None
 
-        timeout = 5
-        try:
-            with gevent.Timeout(timeout):
-                task_manager.schedule()
-        except gevent.Timeout as e:
-            raise AssertionError(f'Premium check query was not completed within {timeout} seconds') from e  # noqa: E501
+        with patch('rotkehlchen.premium.premium.Premium.is_active', MagicMock(return_value=False)):
+            # First run
+            mock_check_premium_status()
+            assert rotki.premium is not None, "First Run"
```

Comment:
If the comment doesn't add value just remove it, is the same as in the assertion. Also if you modify the assert messages you should provide more descriptive information. This applies here and in the other cases. Maybe the first 3 calls can be executed in a for loop to not repeat code
```suggestion
            mock_check_premium_status()
            assert rotki.premium is not None, "Premium object is None and should be active after the first periodic check""
```